src/turnkeyml/run/tensorrt/runtime.py

```diff
@@ -59,18 +59,24 @@ def __init__(
             requires_docker=True,
         )
 
+        self.device_name = self._dynamic_device_name
+
     def _setup(self) -> None:
         # Check if at least one NVIDIA GPU is available locally
-        result = subprocess.run(
-            ["nvidia-smi"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            encoding="utf-8",
-            check=False,
-        )
-
-        if "NVIDIA" not in result.stdout or result.returncode == 1:
-            msg = "No NVIDIA GPUs available on the local machine"
+        msg = "No NVIDIA GPUs available on the local machine"
+        try:
```

Comment:
This was an unrelated bug fix: systems with no Nvidia GPU would crash here when they tried to call nvidia-smi.